File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/TrgMinute.java
package Auxiliary_Modules;

import java.util.Random;

public class TrgMinute extends TrGEV3{
    // trigger true at minute once per hour
    private int hour1 = -1;
    int minute;
    public TrgMinute() {
        Random rand = new Random();
        minute = rand.nextInt(60);
    }

    public TrgMinute(int minute) {
        this.minute = minute;
    }

    @Override
    public Boolean trigger() {
        int tempHour = TimeUtils.getHoursAsInt();
        if(tempHour!=hour1){
            if(TimeUtils.getMinutesAsInt() == minute){
                hour1 = tempHour;
                return true;
            }
        }
        return false;
    }

    @Override
    public void reset() {
        hour1 = -1;
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/TrgMinute.kt
package auxiliary_modules

import java.util.*

class TrgMinute : TrGEV3 {
    // trigger true at minute once per hour
    private var hour1 = -1
    var minute: Int

    constructor() {
        val rand = Random()
        minute = rand.nextInt(60)
    }

    constructor(minute: Int) {
        this.minute = minute
    }

    override fun trigger(): Boolean {
        val tempHour: Int = TimeUtils.hoursAsInt
        if (tempHour != hour1) {
            if (TimeUtils.minutesAsInt == minute) {
                hour1 = tempHour
                return true
            }
        }
        return false
    }

    override fun reset() {
        hour1 = -1
    }
}